From 2b617aa76b9b71ab9cdcaf3948c47e3234e5e8e9 Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Fri, 27 Dec 2019 18:43:00 +0200
Subject: [PATCH] Merge pull request #1573 in SW_UGW/linux from
 bugfix/UGW_SW-46333-sdl-upgrade-operations-have-to-be-carried-on-lower-privileges-841
 to 8.4.1

* commit '91202c2afee4a2fd4b0e68152c5b174621e8293d':
  UGW_SW-46333, UGW_SW-45661, add missing capablility checks in mtd ioctl
---
 drivers/mtd/mtdchar.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mtd/mtdchar.c b/drivers/mtd/mtdchar.c
index 95b6a6640bca..7d9781730713 100644
--- a/drivers/mtd/mtdchar.c
+++ b/drivers/mtd/mtdchar.c
@@ -1063,6 +1063,9 @@ static long mtdchar_unlocked_ioctl(struct file *file, u_int cmd, u_long arg)
 {
 	int ret;
 
+	if (!capable(CAP_SYS_RESOURCE))
+		return -EPERM;
+
 	mutex_lock(&mtd_mutex);
 	ret = mtdchar_ioctl(file, cmd, arg);
 	mutex_unlock(&mtd_mutex);
@@ -1089,6 +1092,9 @@ static long mtdchar_compat_ioctl(struct file *file, unsigned int cmd,
 	void __user *argp = compat_ptr(arg);
 	int ret = 0;
 
+	if (!capable(CAP_SYS_RESOURCE))
+		return -EPERM;
+
 	mutex_lock(&mtd_mutex);
 
 	switch (cmd) {
